core/src/kura.rs

```diff
@@ -692,8 +753,11 @@ type Result<T, E = Error> = std::result::Result<T, E>;
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
     /// Generic IO error
-    #[error("Failed reading/writing from disk")]
-    IO(#[from] std::io::Error),
+    #[error("Failed reading/writing {_1:?} from disk")]
+    IO(#[source] std::io::Error, PathBuf),
+    /// Failed to create the directory
+    #[error("Failed to create the directory")]
+    MkDir(std::io::Error, PathBuf),
```

Comment:
isn't `MkDir` just a subset of `IO`? IMO it should be

Reply:
I don't mind them being together, but I found that working with `io::Error` is a bit difficult. Pattern matching on this error is natural. Casting the trait object inside the `io::Error` is a lot more cumbersome and AFAIK not meant to be done at all.

Reply:
Also, this line shows up because of the conflicts. I'll need to rebase so it disappears.